File: experiments/adrianco/experiment-21-repair-lcm/bookshop/runs/agent=hermes-local_language=elixir_prompt=repair/rep1-failed/book_api/lib/book_api/books/book.ex
defmodule BookApi.Books.Book do
  use Ecto.Schema
  import Ecto.Changeset

  schema "books" do
    field :title, :string
    field :author, :string
    field :year, :integer
    field :isbn, :string

    timestamps()
  end

  @doc false
  def changeset(book, attrs) do
    book
    |> cast(attrs, [:title, :author, :year, :isbn])
    |> validate_required([:title, :author])
    |> validate_length(:title, max: 255)
    |> validate_length(:author, max: 255)
    |> validate_length(:isbn, max: 20)
  end
end

File: experiments/adrianco/experiment-41-repair-80b-fullctx/bookshop/runs/agent=hermes-local_language=java_model=mlxlocal/mlx-community--Qwen3-Coder-Next-4bit_prompt=repair_stack=m80/rep1/src/main/java/com/bookapi/entity/Book.java
package com.bookapi.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

@Entity
@Table(name = "books")
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Title is required")
    @Size(max = 255, message = "Title must be less than 255 characters")
    private String title;

    @NotBlank(message = "Author is required")
    @Size(max = 255, message = "Author must be less than 255 characters")
    private String author;

    @Column(name = "year_of_publication")
    private Integer year;

    @Size(max = 20, message = "ISBN must be less than 20 characters")
    private String isbn;

    // Constructors
    public Book() {}

    public Book(Long id, String title, String author, Integer year, String isbn) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.year = year;
        this.isbn = isbn;
    }

    // Getters
    public Long getId() { return id; }
    public String getTitle() { return title; }
    public String getAuthor() { return author; }
    public Integer getYear() { return year; }
    public String getIsbn() { return isbn; }

    // Setters
    public void setId(Long id) { this.id = id; }
    public void setTitle(String title) { this.title = title; }
    public void setAuthor(String author) { this.author = author; }
    public void setYear(Integer year) { this.year = year; }
    public void setIsbn(String isbn) { this.isbn = isbn; }
}
